lighthouse-core/test/gather/gatherers/source-maps-test.js

```diff
@@ -0,0 +1,211 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env jest */
+
+jest.useFakeTimers();
+
+const Driver = require('../../../gather/driver.js');
+const Connection = require('../../../gather/connections/connection.js');
+const SourceMaps = require('../../../gather/gatherers/source-maps.js');
+const {createMockSendCommandFn, createMockOnFn} = require('../mock-commands.js');
+
+const mapJson = JSON.stringify({
+  version: 3,
+  file: 'out.js',
+  sourceRoot: '',
+  sources: ['foo.js', 'bar.js'],
+  names: ['src', 'maps', 'are', 'fun'],
+  mappings: 'AAgBC,SAAQ,CAAEA',
+});
+
+describe('SourceMaps gatherer', () => {
+  /**
+   * @param {Array<{event: LH.Crdp.Debugger.ScriptParsedEvent, map: string, fetchError: string}>} mapsAndEvents
+   * @return {Promise<LH.Artifacts['SourceMaps']>}
+   */
+  async function runSourceMaps(mapsAndEvents) {
+    const onMock = createMockOnFn();
+
+    const sendCommandMock = createMockSendCommandFn()
+      .mockResponse('Debugger.enable', {})
+      .mockResponse('Debugger.disable', {});
+
+    for (const {map, event, fetchError, resolvedSourceMapUrl} of mapsAndEvents) {
+      onMock.mockEvent('protocolevent', {method: 'Debugger.scriptParsed', params: event});
+
+      if (event.sourceMapURL.startsWith('data:')) {
+        // Only the source maps that need to be fetched use the `evaluateAsync` code path.
+        continue;
+      }
+
+      if (map && fetchError) {
+        throw new Error('should only define map or fetchError, not both.');
+      }
+
+      const value = fetchError ?
+        Object.assign(new Error(), {message: fetchError, __failedInBrowser: true}) :
+        map;
+      sendCommandMock.mockResponse('Runtime.evaluate', {result: {value}}, undefined,
+        ({expression}) => {
+          // Check that the source map url was resolved correctly.
+          // If the test did not define `resolvedSourceMapUrl` explicitly, then use `event.sourceMapURL`.
```

Comment:
i thought it might just be noise but I see it's actually v confusing.

how about just skipping any assert if `resolvedSourceMapUrl` is not defined? adding it to all the tests wouldn't add clarity imo